tools/snp.sh

```diff
@@ -1081,42 +1062,134 @@ attest_guest() {
   ssh_guest_command "sudo insmod /lib/modules/*/kernel/drivers/virt/coco/sev-guest/sev-guest.ko >/dev/null 2>&1 || true"
 
   # Request and display the snp attestation report with random data
-  ssh_guest_command "sudo ./snpguest report attestation-report.bin request-data.txt --random"
-  ssh_guest_command "./snpguest display report attestation-report.bin"
+  if [ "$SVSM" = true ]; then
+	ssh_guest_command "sudo ./snpguest report attestation-report.bin request-data.txt --random"
+  else
+	ssh_guest_command "sudo ./$Home/snpguest/target/release/snpguest report attestation-report.bin request-data.txt --random --vmpl 3"
+  fi
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest display report attestation-report.bin"
 
   # Retrieve ark, ask, vcek (saved in ./certs)
-  ssh_guest_command "./snpguest fetch ca pem ${cpu_code_name} . --endorser vcek"
-  ssh_guest_command "./snpguest fetch vcek pem ${cpu_code_name} . attestation-report.bin"
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest fetch ca pem ${cpu_code_name} ."
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest fetch vcek pem ${cpu_code_name} . attestation-report.bin"
 
   # Verifies that ARK, ASK and VCEK are all properly signed
-  ssh_guest_command "./snpguest verify certs ."
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest verify certs ."
 
   # Verifies the attestation-report trusted compute base matches vcek
-  ssh_guest_command "./snpguest verify attestation . attestation-report.bin"
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest verify attestation . attestation-report.bin"
+  
+ if [ "$SVSM" = true ]; then 
+  
+	  # Use sev-snp-measure utility to calculate the expected measurement
+	  local expected_measurement=$(generate_snp_expected_measurement)
+	  echo -e "\nExpected Measurement (sev-snp-measure):  ${expected_measurement}"
+
+	  # Parse the measurement out of the snp report
+	  local snpguest_report_measurement=$(ssh_guest_command \
+		"./snpguest display report attestation-report.bin \
+		| tr '\n' ' ' \
+		| sed \"s|.*Measurement:\(.*\)Host Data.*|\1\n|g\" \
+		| sed \"s| ||g\"")
+
+	  # Remove any special characters and print the value
+	  snpguest_report_measurement=$(echo ${snpguest_report_measurement} | sed $'s/[^[:print:]\t]//g')
+	  echo -e "Measurement from SNP Attestation Report: ${snpguest_report_measurement}\n"
+
+	  # Compare the expected measurement to the guest report measurement
+	  [[ "${expected_measurement}" == "${snpguest_report_measurement}" ]] \
+		&& echo -e "The expected measurement matches the snp guest report measurement!" \
+		|| { >&2 echo -e "FAIL: measurements do not match"; return 1; }
+else 
+ 	 Verify_measurement
+fi
+}
```

Comment:
Your logic seems backward here. Perhaps check if the variable is set instead of true?